packages/babel-helpers/src/helpers/applyDecs.js

```diff
@@ -537,19 +603,33 @@ function applyClassDecs(ret, targetClass, metadataMap, classDecs) {
   if (classDecs.length > 0) {
     var initializers = [];
     var newClass = targetClass;
-
     var name = targetClass.name;
-    var ctx = Object.assign(
-      {
-        kind: "class",
-        name: name,
-        addInitializer: createAddInitializerMethod(initializers),
-      },
-      createMetadataMethodsForProperty(metadataMap, 0 /* CONSTRUCTOR */, name)
-    );
 
     for (var i = classDecs.length - 1; i >= 0; i--) {
-      var nextNewClass = classDecs[i](newClass, ctx);
+      var decoratorFinishedRef = { v: false };
+
+      try {
+        var ctx = Object.assign(
```

Comment:
Not related to this PR: Object.assign is ES2015, we may need to import `helpers.extends`.